src/v0/destinations/hs/transform.js

```diff
@@ -137,15 +144,31 @@ const processRouterDest = async (inputs, reqMetadata) => {
     }),
   );
 
-  // batch implementation
-  let batchedResponseList = [];
   if (successRespList.length > 0) {
     if (destination.Config.apiVersion === API_VERSION.v3) {
       batchedResponseList = batchEvents(successRespList);
     } else {
       batchedResponseList = legacyBatchEvents(successRespList);
     }
   }
+  return { batchedResponseList, errorRespList };
+};
+// we are batching by default at routerTransform
+const processRouterDest = async (inputs, reqMetadata) => {
+  const tempNewInputs = batchEventsInOrder(inputs);
+  const batchedResponseList = [];
+  const errorRespList = [];
+  const promises = tempNewInputs.map(async (inputEvents) => {
+    const response = await processBatchRouter(inputEvents, reqMetadata);
+    return response;
+  });
+
+  const results = await Promise.all(promises);
```

Comment:
I dont see any specific logic in `processBatchRouter`  how does this solve `userId` level ordering issue?

Reply:
`batchEventsInOrder` function at L158 actually handles the userId level ordering issue.
Yes there is no specific logic in `processBatchRouter`  but `batchEventsInOrder` returns an array of events which should be the right input for router function and separate the functionality for batching and event ordering